src/cmd/services/m3dbnode/main/main_test.go

```diff
@@ -616,3 +634,42 @@ db:
                   endpoint: {{.InitialClusterEndpoint}}
 `
 )
+
+// waitUntilAllShardsAreAvailable continually polls the session checking to see if the topology.Map
+// that the session is currently storing contains a non-zero number of host shard sets, and if so,
+// makes sure that all their shard states are Available.
+func waitUntilAllShardsAreAvailable(t *testing.T, session client.AdminSession) {
+	for {
+		topoMap, err := session.TopologyMap()
+		require.NoError(t, err)
+
+		var (
+			hostShardSets         = topoMap.HostShardSets()
+			allShardsAreAvailable = true
+		)
+
+		if len(hostShardSets) == 0 {
+			// We haven't received an actual topology yet.
+			continue
```

Comment:
Do we want to introduce a `time.Sleep(10 * time.Millisecond)` here as well to prevent a tight loop?

Reply:
good call, I'll just move the loop to the top of the for loop to avoid this in the future